cpp/cert/src/rules/CTR55-CPP/DoNotUseAnAdditiveOperatorOnAnIterator.ql

```diff
@@ -14,24 +14,65 @@
 import cpp
 import codingstandards.cpp.cert
 import codingstandards.cpp.Iterators
+import semmle.code.cpp.controlflow.Dominance
 
-from ContainerIteratorAccess it
+/**
+ * something like:
+ * `end = begin() + size()`
+ */
+Expr calculatedEndCheck(AdditiveOperatorFunctionCall calc) {
+  exists(
+    ContainerAccessWithoutRangeCheck::ContainerSizeCall size,
+    ContainerAccessWithoutRangeCheck::ContainerBeginCall begin
+  |
+    calc.getTarget().hasName("operator+") and
```

Comment:
yeah but that models both + and ++ , which we dont want the ++ operators here, those wouldnt make sense. including them most likely wont fp bc it would capture a more complicated ++ expr with a size and a begin flowing into it, but we still dont want those to count as our idea of sufficient end calculations

Reply:
That make sense, can you create a subclass that captures that to clean it up?